src/main/java/software/amazon/msk/auth/iam/internals/AuthenticationRequestParams.java

```diff
@@ -57,9 +57,14 @@ public String getServiceScope() {
 
     public static AuthenticationRequestParams create(@NonNull String host,
             AWSCredentials credentials,
-            @NonNull String userAgent) throws IllegalArgumentException {
-        Region region = Optional.ofNullable(regionMetadata.tryGetRegionByEndpointDnsSuffix(host))
-                .orElseGet(() -> Regions.getCurrentRegion());
+            @NonNull String userAgent, String awsRegion) throws IllegalArgumentException {
+        Region region;
+        if(awsRegion != null) { // override region if specified
```

Comment:
Can we add a higher level comment to explain how region is picked if specified vs not specified ?